### PR TITLE
[2712] Time-aggregation loss

### LINKED ISSUE #2712: Task: Add time-aggregated loss for physical time evolution
### Describe the task. It can be a feature, documentation, etc.

Producing outputs at a higher temporal frequency than the model’s time-window length now appears to work reasonably well. However, the resulting trajectories sometimes contain undesirable and physically unrealistic jumps within each time window.

This behaviour was observed when using a 6-hour time window with hourly outputs:

Image

A likely explanation is that the output states close to the 6-hour boundaries are more strongly constrained than those near the centre of the window. Consequently, the intermediate states may be less predictable and may evolve inconsistently.

A related problem is discussed in the recent HourGlass preprint, [arXiv:2607.11457](https://arxiv.org/pdf/2607.11457). The paper encourages temporal consistency by applying the training loss not only to the individual output states, but also to temporal statistics and consecutive differences.

### Hedgedoc URL, if you are keeping notes, plots, logs in hedgedoc.

https://gitlab.jsc.fz-juelich.de/hedgedoc/s/tzqiMi8vQ

### Area

- [ ] datasets, data readers, data preparation and transfer
- [x] model
- [x] science
- [ ] infrastructure and engineering
- [ ] evaluation, export and visualization
- [ ] documentation
- [ ] performance

### PR DESCRIPTION
**Pull request by evenmn**
## Description

Time-aggregation loss to avoid jumps within a time window when frequency < window. The ideas are taken from [arxiv:2607.11457](https://arxiv.org/pdf/2607.11457), and the implementation is highly inspired by the Anemoi implementation:

https://github.com/ecmwf/anemoi-core/blob/main/training/src/anemoi/training/losses/aggregate.py

### Invoking the loss

``` bash
train --config ... --options training_config.losses.physical.loss_fcts.mse.time_aggregation_types=[mean,max,min,diff]
```

Screenshot from 2026-08-06 15-14-20

## Issue Number

Closes #2712

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

**Review by clessig — COMMENTED** [review]
Before looking at details, I have some high-level questions: why is `_record_time_agg_candidate()` called in the inner loop of the loss computation. I would have expected that this happens in the contructor of `LossPhysical`. Also, could you give an example for how the config looks like. This doens't seem to be commited?

**Comment by evenmn** [comment]
> Before looking at details, I have some high-level questions: why is `_record_time_agg_candidate()` called in the inner loop of the loss computation. I would have expected that this happens in the contructor of `LossPhysical`. Also, could you give an example for how the config looks like. This doens't seem to be commited?

`_record_time_agg_candidate()` checks if each batch is a candidate for time-aggregation loss. Masking and other things may disable the time-aggregation loss. The function name is a little confusing though, should I replace `record` with something more intuitive? 

This is how to enable the time-aggregation loss. I can push `config_forecasting.yml` adn the stream config as well, it's based on the `atmofs05_6x1h` branch

```
train --config ... --options training_config.losses.physical.loss_fcts.mse.time_aggregation_types=[mean,max,min,diff]
```
